$OpenBSD: patch-mozilla_ipc_chromium_src_chrome_common_ipc_message_utils_h,v 1.3 2012/01/11 01:19:43 nigel Exp $

Add new long long template, may not be best solution.

--- mozilla/ipc/chromium/src/chrome/common/ipc_message_utils.h.orig	Thu Dec 15 01:22:33 2011
+++ mozilla/ipc/chromium/src/chrome/common/ipc_message_utils.h	Mon Dec 19 00:50:51 2011
@@ -193,6 +193,29 @@ struct ParamTraits<unsigned long long> {
     l->append(StringPrintf(L"%ull", p));
   }
 };
+
+template <>
+struct ParamTraits<long long> {
+  typedef long long param_type;
+  static void Write(Message* m, const param_type& p) {
+    m->WriteData(reinterpret_cast<const char*>(&p), sizeof(param_type));
+ }
+  static bool Read(const Message* m, void** iter, param_type* r) {
+    const char *data;
+    int data_size = 0;
+    bool result = m->ReadData(iter, &data, &data_size);
+    if (result && data_size == sizeof(param_type)) {
+      memcpy(r, data, sizeof(param_type));
+    } else {
+      result = false;
+      NOTREACHED();
+    }
+    return result;
+  }
+  static void Log(const param_type& p, std::wstring* l) {
+    l->append(StringPrintf(L"%ull", p));
+  }
+};
 #endif
 
 #if !(defined(OS_MACOSX) || defined(OS_OPENBSD) || defined(OS_WIN) || (defined(OS_LINUX) && defined(ARCH_CPU_64_BITS)))
